### ISSUE TITLE
Implement background class concept for StatScores and derived.

### ISSUE THREAD

**Issue by AndreiSiliUiPath** [Lightning-AI/torchmetrics#1437]
## 🚀 Feature

AFAIK right now there is no concept of a negative/background class in torchmetrics. In `scikit-learn` it is possible to exclude some classes from metric (e.g. f1) calculation. Is this something torchmetrics should cover?

### Motivation

Some problems have a majority negative class that should be ignored. For example in sequence tagging there majority tokens and background and just a few are positive classes (e.g. named entities).

### Additional context
Here's an example how torchmetrics underestimates f1 compared to scikit-learn because the second prediction is counted both as fp for class 0 and as a fn for class 1.

```
import torchmetrics
import torch
import sklearn.metrics
import numpy as np

background_class = 0
num_classes = 3
labels = [x for x in range( num_classes) if x !=  background_class]

Y_true = [2, 1, 0, 0, -1, -1, -1]
Y_predict = [2, 0, 0, 0, 0, 1, 2]

target_np = np.array(Y_true)
pred_np = np.array(Y_predict)

target_th = torch.tensor(Y_true)
pred_th = torch.tensor(Y_predict)

sk_f1 = sklearn.metrics.f1_score(target_np[target_np != -1], pred_np[target_np != -1], labels=labels, average="micro")
f1c_metric = torchmetrics.classification.MulticlassF1Score(average="micro", multidim_average="global",num_classes=num_classes, ignore_index=background_class)
pred_th = torch.where(target_th >= 0, pred_th, 0)
target_th = torch.where(target_th >= 0,target_th, 0)
f1c_metric(pred_th, target_th)
f1c = f1c_metric.compute()
print(round(sk_f1, 4), round(f1c.item(), 4))
>>> 0.6667 0.5
```

Here's an example how torchmetrics overestimates f1 compared to scikit-learn because it does not count the 4th prediction as a fp for class 1.
```
import torchmetrics
import torch
import sklearn.metrics
import numpy as np

background_class = 0
num_classes = 3
labels = [x for x in range( num_classes) if x !=  background_class]

Y_true = [2, 1, 0, 0, -1, -1, -1]
Y_predict = [2, 1, 0, 1, 0, 1, 2]

target_np = np.array(Y_true)
pred_np = np.array(Y_predict)

target_th = torch.tensor(Y_true)
pred_th = torch.tensor(Y_predict)

sk_f1 = sklearn.metrics.f1_score(target_np[target_np != -1], pred_np[target_np != -1], labels=labels, average="micro")
f1c_metric = torchmetrics.classification.MulticlassF1Score(average="micro", multidim_average="global",num_classes=num_classes, ignore_index=background_class)
pred_th = torch.where(target_th >= 0, pred_th, 0)
target_th = torch.where(target_th >= 0,target_th, 0)
f1c_metric(pred_th, target_th)
f1c = f1c_metric.compute()
print(round(sk_f1, 4), round(f1c.item(), 4))
>>> 0.8000 1.0000
```

In torchmetrics 0.9.3 you could achieve the same effect as in scikit-learn by using ignore_index as a background index. Example below:
```
import torchmetrics
import torch
import sklearn.metrics
import numpy as np

background_class = 0
num_classes = 3
labels = [x for x in range( num_classes) if x !=  background_class]

f1_metric = torchmetrics.F1Score(average="micro", mdmc_average="global", num_classes=num_classes, ignore_index=0) # 0.9.3
Y_true = [2, 1, 0, 0]
Y_predict = [2, 1, 0, 1]

target = torch.tensor(Y_true)
preds = torch.tensor(Y_predict)

sk_f1 = sklearn.metrics.f1_score(Y_true, Y_predict, labels=labels, average="micro")
torch_f1 = f1_metric(torch.tensor(Y_predict), torch.tensor(Y_true))

print (sk_f1, torch_f1)
```

For reference, I think the current behaviour of `ignore_index` is correct. The issue is there is no concept of a background class. Here's an extension of Fbeta to support background index.
```
class MulticlassFBetaScoreWithBackground(torchmetrics.classification.MulticlassStatScores):
    is_differentiable: bool = False
    higher_is_better: Optional[bool] = True
    full_state_update: bool = False

    def __init__(
        self,
        beta: float,
        num_classes: int,
        top_k: int = 1,
        average: Optional[Literal["micro", "macro", "weighted", "none"]] = "macro",
        multidim_average: Literal["global", "samplewise"] = "global",
        ignore_index: Optional[int] = None,
        background_index: Optional[int] = None,
        validate_args: bool = True,
        **kwargs: Any,
    ) -> None:
        super().__init__(
            num_classes=num_classes,
            top_k=top_k,
            average="none",
            multidim_average="global",
            ignore_index=ignore_index,
            validate_args=False,
            **kwargs,
        )
        if validate_args:
            torchmetrics.classification.f_beta._multiclass_fbeta_score_arg_validation(beta, num_classes, top_k, "none", "global", ignore_index)
            if background_index is not None and not 0 <= background_index < num_classes:
                raise ValueError(f"Expected argument `background_index` to be an integer between 0 and self.num_classes, but got {background_index}.")
        self.validate_args = validate_args
        self.beta = beta
        self.fbeta_average = average
        self.fbeta_multidim_average = multidim_average
        self.background_index = background_index
        if self.background_index is not None:
            self.register_buffer("positive_indices", torch.tensor([i for i in range(self.num_classes) if i != self.background_index]))

    def compute(self) -> torch.Tensor:
        # get final stats
        tp, fp, tn, fn = self._final_state()
        # remove stats for background class.
        if self.background_index is not None:
            tp = tp[self.positive_indices]
            fp = fp[self.positive_indices]
            tn = tn[self.positive_indices]
            fn = fn[self.positive_indices]
        return torchmetrics.classification.f_beta._fbeta_reduce(tp, fp, tn, fn, self.beta, self.fbeta_average, self.fbeta_multidim_average)


class MulticlassF1ScoreWithBackground(MulticlassFBetaScoreWithBackground):
    is_differentiable: bool = False
    higher_is_better: Optional[bool] = True
    full_state_update: bool = False

    def __init__(
        self,
        num_classes: int,
        top_k: int = 1,
        average: Optional[Literal["micro", "macro", "weighted", "none"]] = "macro",
        multidim_average: Literal["global", "samplewise"] = "global",
        ignore_index: Optional[int] = None,
        background_index: Optional[int] = None,
        validate_args: bool = True,
        **kwargs: Any,
    ) -> None:
        super().__init__(
            beta=1.0,
            num_classes=num_classes,
            top_k=top_k,
            average=average,
            multidim_average=multidim_average,
            ignore_index=ignore_index,
            background_index=background_index,
            validate_args=validate_args,
            **kwargs,
        )
```

**Comment by justusschock** [comment]
What is the difference of ignore_index to background_index in 0.9.3 then?

**Comment by AndreiSiliUiPath** [comment]
As an aside, multiclass_f1_score with micro averaging is equivalent to multiclass_accuracy with micro averaging if you cannot specify a background class.
